mirrord/layer-lib/src/process/windows/execution/pipes.rs

```diff
@@ -0,0 +1,289 @@
+        if self.forwarding_thread.is_some() {
+            return Err(LayerError::IO(std::io::Error::new(
+                std::io::ErrorKind::AlreadyExists,
+                "Forwarding already started".to_string(),
+            )));
+        }
+
+        let src_handle = self.read_handle as usize;
+        let dst_handle = destination as usize;
+
+        let handle = thread::spawn(move || {
+            let mut buffer = vec![0u8; 8192];
+            let mut bytes_read = 0u32;
+            let mut bytes_written = 0u32;
+
+            loop {
+                unsafe {
+                    let read_success = ReadFile(
+                        src_handle as HANDLE,
+                        buffer.as_mut_ptr() as *mut _,
+                        buffer.len() as u32,
+                        &mut bytes_read,
+                        ptr::null_mut(),
+                    );
+
+                    // Check for actual read failure
+                    if read_success == 0 {
+                        let error = GetLastError();
+                        match error {
+                            ERROR_BROKEN_PIPE => {
+                                tracing::debug!(
+                                    "📡 Pipe closed (broken pipe), forwarding thread exiting"
+                                );
+                                break;
+                            }
+                            ERROR_NO_DATA => {
+                                tracing::debug!("📡 No data available, continuing...");
+                                std::thread::sleep(std::time::Duration::from_millis(10));
+                                continue;
+                            }
+                            _ => {
+                                tracing::debug!(
+                                    "📡 ReadFile failed with error {}, forwarding thread exiting",
+                                    error
+                                );
+                                break;
+                            }
+                        }
+                    }
+
+                    // Handle case where ReadFile succeeded but no bytes were read
+                    if bytes_read == 0 {
+                        tracing::debug!("📡 ReadFile succeeded but 0 bytes read, continuing...");
+                        std::thread::sleep(std::time::Duration::from_millis(10));
+                        continue;
+                    }
+
+                    tracing::debug!("📡 Read {} bytes, forwarding...", bytes_read);
+
+                    let write_success = WriteFile(
+                        dst_handle as HANDLE,
+                        buffer.as_ptr() as *const _,
+                        bytes_read,
+                        &mut bytes_written,
+                        ptr::null_mut(),
+                    );
+
+                    if write_success == 0 {
+                        tracing::debug!("📡 Write failed, forwarding thread exiting");
```

Comment:
let's please remove the emojis, they're super inconsistent, and not every font/terminal emulator supports this charset